src/main/java/net/dv8tion/jda/internal/utils/EncodingUtil.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2015-2019 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.internal.utils;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.stream.Collectors;
+
+public class EncodingUtil
+{
+    public static String encodeUTF8(String chars)
+    {
+        try
+        {
+            return URLEncoder.encode(chars, "UTF-8");
+        }
+        catch (UnsupportedEncodingException e)
+        {
+            throw new AssertionError(e); // thanks JDK 1.4
+        }
+    }
+
+    public static String encodeCodepointsUTF8(String input)
+    {
+        if (!input.startsWith("U+"))
```

Comment:
Can codepoints start with `u+`? or must it be `U+`?

Reply:
I have never seen it being lowercase at least. The common notation is `U+CODE`

Reply:
Lets include information about what constitutes a valid format in the message of the `IllegalArgumentException`.
`Invalid Format. Example: "U+1F44D", Provided: "{input}"`

Reply:
`encodeCodepointsUTF8` is only called by `addReactionById` and only if the code starts with `U+`.